es79x/src/main/scala/tech/beshu/ror/es/request/context/types/ResizeEsRequestContext.scala

```
@@ -0,0 +1,62 @@
+ *    along with ReadonlyREST.  If not, see http://www.gnu.org/licenses/
+ */
+package tech.beshu.ror.es.request.context.types
+
+import cats.data.NonEmptyList
+import cats.implicits._
+import org.elasticsearch.action.admin.indices.shrink.ResizeRequest
+import org.elasticsearch.threadpool.ThreadPool
+import tech.beshu.ror.accesscontrol.AccessControlStaticContext
+import tech.beshu.ror.accesscontrol.domain.IndexName
+import tech.beshu.ror.es.RorClusterService
+import tech.beshu.ror.es.request.AclAwareRequestFilter.EsContext
+import tech.beshu.ror.es.request.context.ModificationResult
+import tech.beshu.ror.es.request.context.ModificationResult.{Modified, ShouldBeInterrupted}
+
+class ResizeEsRequestContext(actionRequest: ResizeRequest,
+                             esContext: EsContext,
+                             aclContext: AccessControlStaticContext,
+                             clusterService: RorClusterService,
+                             override val threadPool: ThreadPool)
+  extends BaseIndicesEsRequestContext[ResizeRequest](actionRequest, esContext, aclContext, clusterService, threadPool) {
+
+  override protected def indicesFrom(request: ResizeRequest): Set[IndexName] = {
+    (request.getSourceIndex :: request.getTargetIndexRequest.index() :: Nil)
+      .flatMap(IndexName.fromString)
+      .toSet
+  }
+
+  override protected def update(request: ResizeRequest, filteredIndices: NonEmptyList[IndexName], allAllowedIndices: NonEmptyList[IndexName]): ModificationResult = {
+    val sourceIndex = IndexName.fromString(request.getSourceIndex)
+    val targetIndex = IndexName.fromString(request.getTargetIndexRequest.index())
+
+    val isSourceIndexOnFilteredIndicesList = sourceIndex.exists(filteredIndices.toList.contains(_))
+    val isTargetIndexOnFilteredIndicesList = targetIndex.exists(filteredIndices.toList.contains(_))
+
+    if (isSourceIndexOnFilteredIndicesList && isTargetIndexOnFilteredIndicesList) {
```

according to the comment above:
* it looks like this check is not necessary in general
* this is a write request (not read only), so `indices` rule semantics is as follows: match only when all requested index names are allowed
* this method won't be called if at least one index name, collected in `indicesFrom` method will be forbidden

---

hmm, so maybe we should throw invalid state exception from this method in case it eventually gets called?